rest_framework_idempotency_key/middleware.py

```diff
@@ -20,6 +20,8 @@
 
 
 def get_current_user(request) -> Optional[settings.AUTH_USER_MODEL]:
+    if getattr(settings, 'IDEMPOTENCY_KEY_IGNORE_USER_FIELD', False):
+        return None
     if request.user:
```

Comment:
Making this behavior depends on a setting causes it unchangeable during runtime.
Further, this change has impacts on all the APIs that rely on it.

As discussed with @xeonchen, let's remove `the dependency to request.user`.
That is, let's not care about whether the users between previous and current requests are the same or not.
And no more use of `def get_current_user`, and maybe just store a None into the DB user field.

Reply:
Done